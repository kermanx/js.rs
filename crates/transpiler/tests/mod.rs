use std::fs;

use insta::{assert_snapshot, glob, Settings};
use jsrs_transpiler::transpile;
use oxc::{allocator::Allocator, codegen::Codegen, parser::Parser, span::SourceType};

#[test]
fn test() {
  glob!("fixtures/**/*.jsrs", |path| {
    println!("Testing {}", path.display());
    let input = fs::read_to_string(path).unwrap();
    let mut settings = Settings::clone_current();
    settings.set_omit_expression(true);
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      let output = transpile(&input);
      let allocator = Allocator::new();
      let parsed = Parser::new(&allocator, &output, SourceType::mjs()).parse();
      if parsed.errors.len() > 0 {
        eprintln!("{}", output);
        panic!("{:#?}", parsed.errors);
      }
      let formatted = Codegen::new().build(&parsed.program);
      assert_snapshot!(formatted.code);
    })
  });
}
