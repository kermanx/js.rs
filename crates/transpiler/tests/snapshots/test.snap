---
source: crates/transpiler/tests/mod.rs
input_file: crates/transpiler/tests/fixtures/basic.jsrs
---
export function add(a, b) {
	a + b;
}
export function match_test(x) {
	_match1: {
		const _m0 = a;
		if (_m = matches(_m0, 0)) {
			let [_m0, _m1] = _m0;
			if (_m = matches(_m0, 1)) {
				let [_m0] = _m0;
				var x = _m0;
				if (_m = matches(_m1, 2)) {
					let [_m0] = _m1;
					var y = _m0;
					return x + y;
					break _match1;
				}
			}
		}
		{
			return 0;
		}
	}
}
